### PR TITLE
Dropdownmenu to switch between tabs fix

### PR DESCRIPTION
**Pull request by Craere**
just copy pasted differences from RareSetManager to Stashie

### CODE REVIEW

File: StashieCore.cs [hunk 1]
```diff
@@ -950,7 +950,7 @@ public bool SwitchToTabViaDropdownMenu(StashTabNode tabNode)
                     var brCounter = 0;
 
                     // wait for the dropdown menu to become visible.
-                    while (!dropdownMenu.IsVisible)
+                    while (1 == 2 && !dropdownMenu.IsVisible)
```
Comment (nymann) [inline StashieCore.cs]: This is not a fix, you might as well just remove the while loop.

File: StashieCore.cs [hunk 2]
```diff
@@ -940,7 +940,7 @@ public bool SwitchToTabViaDropdownMenu(StashTabNode tabNode)
                 if (stashPanel.IsVisible && !viewAllTabsButton.IsVisible)
                     return SwitchToTabViaArrowKeys(tabNode);
 
-                var dropdownMenu = _ingameState.ServerData.StashPanel.ViewAllStashPanel;
+                var dropdownMenu = GameController.Game.IngameState.ServerData.StashPanel.ViewAllStashPanel;
```
Comment (nymann) [inline StashieCore.cs]: This change adds nothing.

File: StashieCore.cs [hunk 3]
```diff
@@ -961,12 +961,10 @@ public bool SwitchToTabViaDropdownMenu(StashTabNode tabNode)
...
                     // Make sure that we are scrolled to the top in the menu.
-                    if (_ingameState.ServerData.StashPanel.TotalStashes > 30)
+                    if (GameController.Game.IngameState.ServerData.StashPanel.TotalStashes > 30)
                     {
-                        Thread.Sleep(WHILE_DELAY);
-                        Mouse.VerticalScroll(true, 10);
-                        Thread.Sleep(WHILE_DELAY);
-                        Mouse.VerticalScroll(true, 10);
+                        Mouse.VerticalScroll(true, 5);
+                        Thread.Sleep(latency + Settings.ExtraDelay);
```
Comment (nymann) [inline StashieCore.cs]: For untracked mouse movements, there shouldn't be an extra delay.

File: StashieCore.cs [hunk 4]
```diff
@@ -975,9 +973,9 @@ public bool SwitchToTabViaDropdownMenu(StashTabNode tabNode)
                 // 0 is the icon (fx. chaos orb).
                 // 1 is the name of the tab.
                 // 2 is the slider.
-                var slider = dropdownMenu.Children[1].ChildCount == _ingameState.ServerData.StashPanel.TotalStashes;
-
-                var noSlider = dropdownMenu.Children[2].ChildCount == _ingameState.ServerData.StashPanel.TotalStashes;
+                var totalStashes = GameController.Game.IngameState.ServerData.StashPanel.TotalStashes;
+                var slider = dropdownMenu.Children[1].ChildCount == totalStashes;
```
Comment (nymann) [inline StashieCore.cs]: Nothing changed.

File: StashieCore.cs [hunk 5]
```diff
@@ -995,7 +993,7 @@ public bool SwitchToTabViaDropdownMenu(StashTabNode tabNode)
                 }
 
                 Mouse.SetCursorPosAndLeftClick(tabPos.Center, Settings.ExtraDelay, _windowOffset);
-                Thread.Sleep(latency);
+                Thread.Sleep(latency + Settings.ExtraDelay);
```
Comment (nymann) [inline StashieCore.cs]: This click action is client side, and therefore doesn't need the extra delay.

**Comment by zaafar** [comment]
@nymann finally someone who actually review the PR before merge!. :)

**Pull request closed without merging**